Makefile

```diff
@@ -12,88 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# The binary to build (just the basename).
-BIN ?= $(wildcard velero-*)
-
-# This repo's root import path (under GOPATH).
 PKG := github.com/vmware-tanzu/velero-plugin-example
+BIN := velero-plugin-example
 
-BUILD_IMAGE ?= golang:1.12-stretch
-
-IMAGE ?= velero/velero-plugin-example
+REGISTRY ?= velero
+IMAGE    ?= $(REGISTRY)/velero-plugin-example
+VERSION  ?= master 
 
-# Which architecture to build - see $(ALL_ARCH) for options.
-# if the 'local' rule is being run, detect the ARCH from 'go env'
+# Which architecture to build.
+# if the 'local' rule is being run, detect the GOOS/GOARCH from 'go env'
 # if it wasn't specified by the caller.
-local : ARCH ?= $(shell go env GOOS)-$(shell go env GOARCH)
-ARCH ?= linux-amd64
-
-platform_temp = $(subst -, ,$(ARCH))
-GOOS = $(word 1, $(platform_temp))
-GOARCH = $(word 2, $(platform_temp))
-
-all: $(addprefix build-, $(BIN))
+local: GOOS ?= $(shell go env GOOS)
+GOOS ?= linux
 
-build-%:
-	$(MAKE) --no-print-directory BIN=$* build
+local: GOARCH ?= $(shell go env GOARCH)
+GOARCH ?= amd64
 
+# local builds the binary using 'go build' in the local environment.
+.PHONY: local
 local: build-dirs
-	GOOS=$(GOOS) \
-	GOARCH=$(GOARCH) \
-	PKG=$(PKG) \
-	BIN=$(BIN) \
-	OUTPUT_DIR=$$(pwd)/_output/bin/$(GOOS)/$(GOARCH) \
-	./hack/build.sh
-
-build: _output/bin/$(GOOS)/$(GOARCH)/$(BIN)
-
-_output/bin/$(GOOS)/$(GOARCH)/$(BIN): build-dirs
-	@echo "building: $@"
-	$(MAKE) shell CMD="-c '\
-		GOOS=$(GOOS) \
-		GOARCH=$(GOARCH) \
-		PKG=$(PKG) \
-		BIN=$(BIN) \
-		OUTPUT_DIR=/output/$(GOOS)/$(GOARCH) \
-		./hack/build.sh'"
-
-TTY := $(shell tty -s && echo "-t")
-
-shell: build-dirs 
-	@echo "running docker: $@"
-	@docker run \
-		-e GOFLAGS \
-		-i $(TTY) \
-		--rm \
-		-u $$(id -u):$$(id -g) \
-		-v $$(pwd)/.go/pkg:/go/pkg \
-		-v $$(pwd)/.go/src:/go/src \
-		-v $$(pwd)/.go/std:/go/std \
-		-v $$(pwd):/go/src/$(PKG) \
-		-v $$(pwd)/.go/std/$(GOOS)_$(GOARCH):/usr/local/go/pkg/$(GOOS)_$(GOARCH)_static \
-		-v "$$(pwd)/.go/go-build:/.cache/go-build:delegated" \
-		-e CGO_ENABLED=0 \
-		-w /go/src/$(PKG) \
-		$(BUILD_IMAGE) \
-		go build -installsuffix "static" -i -v -o _output/bin/$(GOOS)/$(GOARCH)/$(BIN) ./$(BIN)
-
+	CGO_ENABLED=0 go build -v -o _output/bin/$(GOOS)/$(GOARCH) .
+
+# test runs unit tests using 'go test' in the local environment.
+.PHONY: test
+test:
+	CGO_ENABLED=0 go test -v -timeout 60s ./...
+
+# ci is a convenience target for CI builds.
+.PHONY: ci
+ci: local test
+
+# container builds a Docker image containing the binary.
+.PHONY: container
+container:
+	docker build -t $(IMAGE):$(VERSION) .
+
+# push pushes the Docker image to its registry.
+.PHONY: push
+push:
+	@docker push $(IMAGE):$(VERSION)
+ifeq ($(TAG_LATEST), true)
```

Comment:
If this is condition is `true` will we be pushing the image twice?

Reply:
yes, with two different tags, e.g. `velero/velero-plugin-example:master` and `velero/velero-plugin-example:latest`. The content will be identical.